cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisionerTable.java

```diff
@@ -120,25 +103,27 @@ public void putTaskInfo(ProvisioningTaskInfo taskInfo) throws IOException {
    */
   public void deleteTaskInfo(ProgramRunId runId) throws IOException {
 
-    // delete provision and deprovision tasks
-    table.delete(ImmutableList.of
-      (Fields.stringField(StoreDefinition.ProvisionerStore.NAMESPACE_FIELD, runId.getNamespace()),
-       Fields.stringField(StoreDefinition.ProvisionerStore.APPLICATION_FIELD, runId.getApplication()),
-       Fields.stringField(StoreDefinition.ProvisionerStore.VERSION_FIELD, runId.getVersion()),
-       Fields.stringField(StoreDefinition.ProvisionerStore.PROGRAM_TYPE_FIELD, runId.getType().name()),
-       Fields.stringField(StoreDefinition.ProvisionerStore.PROGRAM_FIELD, runId.getProgram()),
-       Fields.stringField(StoreDefinition.ProvisionerStore.RUN_FIELD, runId.getRun()),
-       Fields.stringField(StoreDefinition.ProvisionerStore.KEY_TYPE, ProvisioningOp.Type.PROVISION.name())));
+    // Delete the keys with Provision and Deprovision
+    table.deleteAll(Range.from(
```

Comment:
this needs to be a singleton range. What you have here will delete all other tasks that have a greater key.

Reply:
added. thanks for the catch.